common/kvstore/src/main/java/org/apache/spark/util/kvstore/RocksDB.java

```diff
@@ -355,26 +355,21 @@ public void close() throws IOException {
     }
   }
 
-  /**
-   * Closes the given iterator if the DB is still open. Trying to close a JNI RocksDB handle
-   * with a closed DB can cause JVM crashes, so this ensures that situation does not happen.
-   */
-  void closeIterator(RocksDBIterator<?> it) throws IOException {
-    notifyIteratorClosed(it);
-    synchronized (this._db) {
-      org.rocksdb.RocksDB _db = this._db.get();
-      if (_db != null) {
-        it.close();
-      }
-    }
+  public AtomicReference<org.rocksdb.RocksDB> getRocksDB() {
+    return _db;
   }
 
-  /**
-   * Remove iterator from iterator tracker. `RocksDBIterator` calls it to notify
-   * iterator is closed.
-   */
-  void notifyIteratorClosed(RocksDBIterator<?> it) {
-    iteratorTracker.removeIf(ref -> it.equals(ref.get()));
+  public ConcurrentLinkedQueue<Reference<RocksDBIterator<?>>> getIteratorTracker() {
+    return iteratorTracker;
+  }
+
+  public Optional<Reference<RocksDBIterator<?>>> iteratorReference(RocksDBIterator<?> rocksDBIterator) {
```

Comment:
ditto

Reply:
This method should add `@VisibleForTesting`

Reply:
The method name is something I came up with on the spot, there might be a more suitable one.